dirsrvtests/tests/suites/indexes/regression_test.py

```diff
@@ -759,6 +759,129 @@ def test_update_eq_index_after_deleting_and_readding_attribute_in_one_step(topo)
         assert False
 
 
+def test_concurrent_modifications_during_indexing(topo):
+    """Test that concurrent modifications are rejected during index rebuilding
+
+    :id: 1df16d5a-1b92-46b7-8435-876b87545748
+    :setup: Standalone Instance
+    :steps:
+        1. Create homeDirectory index with substring matching
+        2. Create 100 users with large homeDirectory values to slow down indexing
+        3. Start an indexing task WITHOUT waiting for its completion
+        4. Monitor task status and verify modifications are rejected with UNWILLING_TO_PERFORM
+        5. Verify modifications work once indexing is complete
+    :expectedresults:
+        1. Index configuration succeeds
+        2. Users are successfully created
+        3. Indexing task starts
+        4. Modifications return UNWILLING_TO_PERFORM while indexing is in progress
+        5. Modifications succeed once indexing is complete
+    """
+
+    inst = topo.standalone
+
+    backends = Backends(inst)
+    backend = backends.get(DEFAULT_BENAME)
+    indexes = backend.get_indexes()
+
+    try:
```

Comment:
I would move this to fixture, since we ensure at the start and the end of the test that this index is not present.